llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3025,6 +3025,9 @@ static void combineMetadata(Instruction *K, const Instruction *J,
         // Preserve !nosanitize if both K and J have it.
         K->setMetadata(Kind, JMD);
         break;
+      case LLVMContext::MD_captures:
+        K->setMetadata(Kind, JMD ? MDNode::concatenate(JMD, KMD) : nullptr);
```

Comment:
`address_is_null + address` can be `address`. It may be better to add a `StoreInst::setCaptureComponents` helper to update MD nodes. Then we can guarantee the list is always canonical.